### PR TITLE
Task #2567 - Sign up

### PR DESCRIPTION
**Pull request by ndviet2307**
https://edu-redmine.framgia.vn/issues/2567

### CODE REVIEW

**Comment by ndviet2307** [comment]
screenshot from 2018-10-12 11-31-09

**Comment by ndviet2307** [comment]
ready

**Comment by DuongNT1512** [comment]
pull của em còn chưa có ai check nhé

**Comment by DuongNT1512** [comment]
commented

File: app/helpers/devise_helper.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          <ul>#{messages}</ul>
+        </div>
+      </div>
+    HTML
+
+    html.html_safe
+  end
+
+  def devise_error_messages?
+    !resource.errors.empty?
```
Comment (DuongNT1512) [inline app/helpers/devise_helper.rb]: resource.errors.any?

File: app/views/users/confirmations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<h2 class="text-center mt-3">
+  <%= t ".resend_confirm" %>
+</h2>
+<div class="container col-md-4 my-3">
+  <%= form_for resource, as: resource_name,
+    url: confirmation_path(resource_name), html: {method: :post} do |f| %>
+  <%= devise_error_messages! %>
```
Comment (DuongNT1512) [inline app/views/users/confirmations/new.html.erb]: indent

File: app/views/users/passwords/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<h2 class="text-center mt-3">
+  <%= t ".forgot_password" %>
+</h2>
+<div class="container col-md-4 my-3">
+  <%= form_for resource, as: resource_name,
+    url: password_path(resource_name), html: {method: :post} do |f| %>
+  <%= devise_error_messages! %>
```
Comment (DuongNT1512) [inline app/views/users/passwords/new.html.erb]: tương tự

**Comment by DuongNT1512** [comment]
commented

**Comment by ndviet2307** [comment]
screenshot from 2018-10-12 14-39-28

**Comment by ndviet2307** [comment]
ready

File: app/helpers/devise_helper.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module DeviseHelper
+  def devise_error_messages!
+    return "" unless devise_error_messages?
+
+    messages = resource.errors.full_messages.map do |msg|
+      content_tag(:li, msg)
```
Comment (thanhhungchu95) [inline app/helpers/devise_helper.rb]: bỏ ()

**Comment by thanhhungchu95** [comment]
commented

**Comment by ndviet2307** [comment]
screenshot from 2018-10-12 14-54-25

**Comment by ndviet2307** [comment]
ready

**Pull request merged**